Content.Server/Zombies/ZombieSystem.cs

```diff
@@ -74,18 +81,47 @@ public override void Update(float frameTime)
                 // 1x at 30s, 3x at 60s, 6x at 90s, 10x at 120s.
                 var pain_multiple = 0.1 + 0.02 * comp.InfectedSecs + 0.0005 * comp.InfectedSecs * comp.InfectedSecs;
                 comp.NextTick = curTime;
-                _damageable.TryChangeDamage(uid, comp.Damage * pain_multiple, true, false);
+                _damageable.TryChangeDamage(uid, comp.Damage * pain_multiple, true, false, damage);
             }
 
-            var zomb_query = EntityQueryEnumerator<ZombieComponent>();
             // Heal the zombified
-            while (zomb_query.MoveNext(out var uid, out var comp))
+            while (zombQuery.MoveNext(out var uid, out var comp, out var damage, out var mobState))
             {
+                // Process only once per second
                 if (comp.NextTick + TimeSpan.FromSeconds(1) > curTime)
                     continue;
 
                 comp.NextTick = curTime;
-                _damageable.TryChangeDamage(uid, comp.Damage, true, false);
+
+                // Healing increases over time for zombies currently in crit
+                if (mobState.CurrentState == MobState.Critical)
+                {
+                    comp.SecondsCrit += 1;
+
+                    // Roll to see if this zombie is not coming back.
+                    //   Note that due to damage reductions it takes a lot of hits to gib a zombie without this.
+                    if (damage.TotalDamage > _random.NextFloat(130.0f, 200.0f))
+                    {
+                        // You're dead! No reviving for you.
+                        _mobThreshold.SetAllowRevives(uid, false);
+                        _damageable.TryChangeDamage(uid, comp.ForceDeathDamage, true, false, damage);
+                        continue;
+                    }
+                }
+                else if (mobState.CurrentState == MobState.Dead)
+                {
+                    // No healing
+                    continue;
```

Comment:
i think it would be better to just subscribe to the mob changed event and have like a 20% rolled once every time you go in crit that if met, gibs you. Zombify should also reduce the gib damage trigger threshold to like 200/150, i.e
![image](https://github.com/space-wizards/space-station-14/assets/60196617/852342da-ce84-487c-8342-134dd6e2f0ef)

Reply:
With "roll to kill zombies with high damage" The goal was that zombies ALWAYS respawn if you kill them but don't deal more damage. But then some extra damage is likely (but not guaranteed) to kill them for real.
I liked how it is semi-random so you can be confident, but not quite sure that the zombie is down forever
The zombie you think is dead that jumps up to scare you is funny, in controlled amounts
Also, dead zombies can be cloned back to regular crew. If we force players to always gib their crewmates then they cannot clone them
But I'm sensitive that Zombies have very high damage resistance so I didn't want the kill for real threshold to be too high.
What complicates these constants like 130 and 200 is that they are hardcoded for humans. I need a formula that calculates good values for animal zombies and dragons etc.